Gems/WhiteBox/Code/Source/SubComponentModes/EditorWhiteBoxDefaultMode.cpp

```diff
@@ -132,6 +148,166 @@ namespace WhiteBox
         EditorWhiteBoxDefaultModeRequestBus::Handler::BusDisconnect();
     }
 
+    void DefaultMode::RegisterActionUpdaters()
+    {
+        auto actionManagerInterface = AZ::Interface<AzToolsFramework::ActionManagerInterface>::Get();
+        AZ_Assert(actionManagerInterface, "WhiteBoxDefaultMode - could not get ActionManagerInterface on RegisterActionUpdaters.");
+
+        actionManagerInterface->RegisterActionUpdater(WhiteBoxDefaultSelectionChangeUpdaterIdentifier);
+    }
+
+    void DefaultMode::RegisterActions()
+    {
+        auto actionManagerInterface = AZ::Interface<AzToolsFramework::ActionManagerInterface>::Get();
+        AZ_Assert(actionManagerInterface, "WhiteBoxDefaultMode - could not get ActionManagerInterface on RegisterActions.");
+
+        auto hotKeyManagerInterface = AZ::Interface<AzToolsFramework::HotKeyManagerInterface>::Get();
+        AZ_Assert(hotKeyManagerInterface, "WhiteBoxDefaultMode - could not get HotKeyManagerInterface on RegisterActions.");
+
+        // Hide Edge
+        {
+            constexpr AZStd::string_view actionIdentifier = "o3de.action.whiteBoxComponentMode.Default.hideEdge";
+            AzToolsFramework::ActionProperties actionProperties;
+            actionProperties.m_name = HideEdgeTitle;
+            actionProperties.m_description = HideEdgeDesc;
+            actionProperties.m_category = "White Box Component Mode - Default";
+
+            actionManagerInterface->RegisterAction(
+                EditorMainWindowActionContextIdentifier,
+                actionIdentifier,
+                actionProperties,
+                []
+                {
+                    auto componentModeCollectionInterface = AZ::Interface<AzToolsFramework::ComponentModeCollectionInterface>::Get();
+                    AZ_Assert(componentModeCollectionInterface, "Could not retrieve component mode collection.");
+
+                    componentModeCollectionInterface->EnumerateActiveComponents(
+                        [](const AZ::EntityComponentIdPair& entityComponentIdPair, const AZ::Uuid&)
+                        {
+                            EditorWhiteBoxDefaultModeRequestBus::Event(
+                                entityComponentIdPair, &EditorWhiteBoxDefaultModeRequests::HideSelectedEdge);
+                        }
+                    );
+                }
+            );
+
+            actionManagerInterface->InstallEnabledStateCallback(
+                actionIdentifier,
+                [&]()
```

Comment:
suggestion: It's generally preferred to capture variables explicitly instead of a blanket reference capture, does this add too much noise or is it doable here? Might be nice to capture this and whatever other local variables are needed (if any)

Reply:
It was actually unnecessary, fixed.